frontend/packages/dev-console/src/components/import/import-validation-utils.ts

```diff
@@ -37,13 +37,13 @@ export const detectGitType = (url: string): string => {
     // Not a URL
     return GitTypes.invalid;
   }
-  if (url.includes('github.com')) {
+  if (url.includes('https://github.com/') || url.includes('@github.com:')) {
     return GitTypes.github;
   }
-  if (url.includes('bitbucket.org')) {
+  if (url.includes('https://bitbucket.org/') || url.includes('@bitbucket.org:')) {
     return GitTypes.bitbucket;
   }
-  if (url.includes('gitlab.com')) {
+  if (url.includes('https://gitlab.com/') || url.includes('@gitlab.com:')) {
```

Comment:
I'm not a fan of repeating code with one minor difference, this makes it harder to maintain and if we need to add more checks, it's a bit of a mess.

If you rebase on top of master, you'll get #3429 and perhaps you can just add the prefix/suffix check to the renamed `urlRegex` (now `gitUrlRegex`).

Reply:
Correct @christianvogt - I do wonder if there is something better we can do here, but that is definitely the bare min in my mind to make this acceptable. Saves on repeat and gives us a clean spot to expand the logic uniformly (if the future desires it).